exercises.py

```diff
@@ -1,13 +1,394 @@
+it is destroyed
+
+# 7. What is a return value? Can a return value be part of an expression?
+the value that a function call evaluates.  yes, because expressions are composed of values and operators. and a return value is a value
+
+# 8. If a function does not have a return statement, what is the return value of a call to that function?
+it returns nothing, with value of None
+
+# 9. How can you force a variable in a function to refer to the global variable?
+by not using it locally in an assignment statement
+
+# 10. What is the data type of None?
+NoneType
+
+# 11. What does the import areallyourpetsnamederic statement do?
+it would import a module called areallyourpetsnamederic, if it existed
+
+# 12. If you had a function named bacon() in a module named spam, how would you call it after importing spam?
+import spam
+
+spam.bacon()
+
+# 13. How can you prevent a program from crashing when it gets an error?
+with try and except statements, by putting the code that may have an error in a try clause
+
+# 14. What goes in the try clause? What goes in the except clause?
+the code that may have an error goes in the try clause
+what to do if that error happens goes in the except clause
+
+################################################################################################
+
+
+
+
+
+
 # Chapter 4
 
+# 1. What is []?
+an empty list
+
+# 2. How would you assign the value 'hello' as the third value in a list stored in a variable named spam? (Assume spam contains [2, 4, 6, 8, 10].)
+spam.insert(2,"hello")
+
+# For the following three questions, let’s say spam contains the list ['a', 'b', 'c', 'd'].
+# 3. What does spam[int(int('3' * 2) // 11)] evaluate to?
+d
+
+# 4. What does spam[-1] evaluate to?
+d
+
+# 5. What does spam[:2] evaluate to?
+['a', 'b']
+
+# For the following three questions, let’s say bacon contains the list [3.14, 'cat', 11, 'cat', True].
+# 6. What does bacon.index('cat') evaluate to?
+1
+
+# 7. What does bacon.append(99) make the list value in bacon look like?
+[3.14, 'cat', 11, 'cat', True, 99]
+
+# 8. What does bacon.remove('cat') make the list value in bacon look like?
+[3.14, 11, 'cat', True, 99]
+
+# 9. What are the operators for list concatenation and list replication?
++ for concatenation, and * for replication
+
+# 10. What is the difference between the append() and insert() list methods?
+.append() adds something to the end of a list, and .insert ( , ) adds something to an indexed spot in the list
+
+# 11. What are two ways to remove values from a list?
+del list_name[index value] and list_name.remove()
+
+# 12. Name a few ways that list values are similar to string values.
+They both start and end with something string values use "", list values use []
+they are both "lists" of single text characters
+
+# 13. What is the difference between lists and tuples?
+tuples use () instead of []
+tuples contain 2+ items of related data (thing1, thing2, thing 3)
```

Comment:
Can a tuple contain one item? What character does a tuple have to contain to make it a tuple?